W&B Run: https://app.wandb.ai/gvaihir_ucsf/image_sudoku/runs/dn8b4ass
>>> autoencoder, discriminator, generator, encoder, decoder = create_model(input_dim=input_dim,
...                                                                            latent_dim=latent_dim,
...                                                                            verbose=True, save_graph=False,
...                                                                            conv=conv,
...                                                                            adversarial=adversarial)
WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:74: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:517: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:4138: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:3976: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:174: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:181: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

2019-09-04 07:37:03.567720: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-09-04 07:37:03.574966: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2300075000 Hz
2019-09-04 07:37:03.575715: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55fb370bcff0 executing computations on platform Host. Devices:
2019-09-04 07:37:03.575745: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2019-09-04 07:37:03.578352: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
2019-09-04 07:37:03.647963: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-09-04 07:37:03.648592: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55fb371710c0 executing computations on platform CUDA. Devices:
2019-09-04 07:37:03.648625: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Tesla M60, Compute Capability 5.2
2019-09-04 07:37:03.648813: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-09-04 07:37:03.649360: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: Tesla M60 major: 5 minor: 2 memoryClockRate(GHz): 1.1775
pciBusID: 0000:00:1e.0
2019-09-04 07:37:03.649625: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-09-04 07:37:03.651152: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
2019-09-04 07:37:03.652441: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
2019-09-04 07:37:03.652730: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
2019-09-04 07:37:03.654450: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
2019-09-04 07:37:03.655741: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
2019-09-04 07:37:03.655862: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcudnn.so.7'; dlerror: libcudnn.so.7: cannot open shared object file: No such file or directory
2019-09-04 07:37:03.655882: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1663] Cannot dlopen some GPU libraries. Skipping registering GPU devices...
2019-09-04 07:37:03.655911: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-09-04 07:37:03.655925: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 
2019-09-04 07:37:03.655932: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N 
2019-09-04 07:37:03.687390: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.
WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:1834: The name tf.nn.fused_batch_norm is deprecated. Please use tf.compat.v1.nn.fused_batch_norm instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:2018: The name tf.image.resize_nearest_neighbor is deprecated. Please use tf.compat.v1.image.resize_nearest_neighbor instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/optimizers.py:790: The name tf.train.Optimizer is deprecated. Please use tf.compat.v1.train.Optimizer instead.

WARNING:tensorflow:From /home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/tensorflow/python/ops/nn_impl.py:180: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
Autoencoder Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         (None, 104, 104, 3)       0         
_________________________________________________________________
sequential_1 (Sequential)    (None, 32)                22069760  
_________________________________________________________________
sequential_2 (Sequential)    (None, 104, 104, 3)       22053091  
=================================================================
Total params: 44,122,851
Trainable params: 44,111,843
Non-trainable params: 11,008
_________________________________________________________________
None
Discriminator Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
dense_7 (Dense)              (None, 2704)              89232     
_________________________________________________________________
dense_8 (Dense)              (None, 2704)              7314320   
_________________________________________________________________
dense_9 (Dense)              (None, 1)                 2705      
=================================================================
/home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/engine/training.py:490: UserWarning: Discrepancy between trainable weights and collected trainable weights, did you set `model.trainable` without calling `model.compile` after ?
  'Discrepancy between trainable weights and collected trainable'
Total params: 14,812,514
Trainable params: 7,406,257
Non-trainable params: 7,406,257
_________________________________________________________________
None
Generator Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_2 (InputLayer)         (None, 104, 104, 3)       0         
_________________________________________________________________
sequential_1 (Sequential)    (None, 32)                22069760  
_________________________________________________________________
sequential_3 (Sequential)    (None, 1)                 7406257   
=================================================================
Total params: 29,476,017
Trainable params: 22,058,752
Non-trainable params: 7,417,265
_________________________________________________________________
None
>>> 
>>> data_loader = ImageDataGenerator(
...     featurewise_center=True,
...     featurewise_std_normalization=True,
...     shear_range=0.2,
...     zoom_range=0.2,
...     horizontal_flip=True)
>>> 
>>> train_data = data_loader.flow_from_directory(
...     img_wd,
...     target_size=(input_dim[0], input_dim[0]),
...     batch_size=batch,
...     class_mode='input')
Found 1083 images belonging to 1 classes.
>>> train(train_data=train_data, out=out,
...     latent_dim=latent_dim, n_epochs=epoch,
...     autoencoder=autoencoder, discriminator=discriminator,
...     generator=generator, encoder=encoder, decoder=decoder,
...     adversarial=adversarial)
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "<stdin>", line 22, in train
>>>   File "/home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/legacy/interfaces.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "/home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/engine/training.py", line 1418, in fit_generator
    initial_epoch=initial_epoch)
  File "/home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/engine/training_generator.py", line 55, in fit_generator
    raise ValueError('`steps_per_epoch=None` is only valid for a'
ValueError: `steps_per_epoch=None` is only valid for a generator based on the `keras.utils.Sequence` class. Please specify `steps_per_epoch` or use the `keras.utils.Sequence` class.
latent_dim
32
>>> batch
64
>>> batch=56
>>> img_wd = "/home/aogorodnikov/data_aae/"
>>> batch = 64
>>> input_dim = (104,104,3)
>>> latent_dim = 32
>>> epoch = 10
>>> conv=True
>>> adversarial=True
>>> out="/home/aogorodnikov/aae/"
>>> # initialize monitoring with WandB
... wandb.init()
W&B Run: https://app.wandb.ai/gvaihir_ucsf/image_sudoku/runs/dn8b4ass
>>> 
>>> 
>>> autoencoder, discriminator, generator, encoder, decoder = create_model(input_dim=input_dim,
...                                                                            latent_dim=latent_dim,
...                                                                            verbose=True, save_graph=False,
...                                                                            conv=conv,
...                                                                            adversarial=adversarial)
Autoencoder Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_3 (InputLayer)         (None, 104, 104, 3)       0         
_________________________________________________________________
sequential_4 (Sequential)    (None, 32)                22069760  
_________________________________________________________________
sequential_5 (Sequential)    (None, 104, 104, 3)       22053091  
=================================================================
Total params: 44,122,851
Trainable params: 44,111,843
Non-trainable params: 11,008
_________________________________________________________________
None
Discriminator Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
dense_16 (Dense)             (None, 2704)              89232     
_________________________________________________________________
dense_17 (Dense)             (None, 2704)              7314320   
_________________________________________________________________
dense_18 (Dense)             (None, 1)                 2705      
=================================================================
/home/aogorodnikov/anaconda3/envs/imgSudoku/lib/python3.7/site-packages/keras/engine/training.py:490: UserWarning: Discrepancy between trainable weights and collected trainable weights, did you set `model.trainable` without calling `model.compile` after ?
  'Discrepancy between trainable weights and collected trainable'
Total params: 14,812,514
Trainable params: 7,406,257
Non-trainable params: 7,406,257
_________________________________________________________________
None
Generator Architecture
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_4 (InputLayer)         (None, 104, 104, 3)       0         
_________________________________________________________________
sequential_4 (Sequential)    (None, 32)                22069760  
_________________________________________________________________
sequential_6 (Sequential)    (None, 1)                 7406257   
=================================================================
Total params: 29,476,017
Trainable params: 22,058,752
Non-trainable params: 7,417,265
_________________________________________________________________
None
>>> 
>>> # LOAD DATA
... data_loader = ImageDataGenerator(
...     featurewise_center=True,
...     featurewise_std_normalization=True,
...     shear_range=0.2,
...     zoom_range=0.2,
...     horizontal_flip=True)
>>> 
>>> train_data = data_loader.flow_from_directory(
...     img_wd,
...     target_size=(input_dim[0], input_dim[0]),
...     batch_size=batch,
...     class_mode='input')
Found 1083 images belonging to 1 classes.
>>> Found 1083 images belonging to 1 classes.
KeyboardInterrupt
